modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -136,9 +136,43 @@ public void test() throws Exception {
 			HTTPTestUtil.invokeHttpCode(null, endpointPath2, Http.Method.GET));
 	}
 
+	@Test
+	public void testEndpointPagination() throws Exception {
+		int page = 2;
+
+		int pageSize = 5;
+
+		int totalCount = 25;
+
+		int lastPage = -Math.floorDiv(-totalCount, pageSize);
+
+		for (int i = 0; i < totalCount; i++) {
+			_createCustomObjectEntry(
+				_objectDefinitionJSONObject, RandomTestUtil.randomString());
+		}
+
+		APIApplicationPublisherUtil.publishApplications(
+			_apiApplicationPublisher,
+			HeadlessBuilderResourceTest._apiApplication1);
+
+		JSONObject jsonObject = HTTPTestUtil.invoke(
+			null,
+			String.format(
+				"%s?page=%d&pageSize=%d",
+				_apiApplication1.getBaseURL() + _API_APPLICATION_PATH_1, page,
+				pageSize),
+			Http.Method.GET);
+
+		Assert.assertEquals(lastPage, jsonObject.get("lastPage"));
```

Comment:
I think we could use `JSONAssert.assertEquals` instead as it contains the `JSONCompareMode` argument to modify the way we want to compare (for example, without having into account the order)

Reply:
Done